File: summer_school/turtle.cpp
#include<iostream>
#include<vector>

#define MOD 1000000007

using namespace std;

int main(){
    long long n;
    cin >> n;
    char matrix[n][n];
    long long dp[n][n];
    for(long long i = 0 ; i<n; i++){
        for(long long j = 0 ; j < n ; j++){
            cin >> matrix[i][j];
            dp[i][j]=0;
        }}
    bool flag = false;
    for(long long i = 0 ; i < n; i++){
        if(!flag){ 
            dp[i][0] = 1;
        }
        else{
            dp[i][0] = 0;
        }
        if(matrix[i][0] == '*'){
            flag = true;
            dp[i][0] = 0;
        }
    }
    flag = false;
    for(long long i = 0 ; i < n; i++){
        if(!flag){ 
            dp[0][i] = 1;
        }
        else{
            dp[0][i] = 0;
        }
        if(matrix[0][i] == '*'){
            flag = true;
            dp[0][i] = 0;
        }
    }
    for(long long i = 1; i< n; i++){
        for(long long j = 1; j < n; j++){
            if(matrix[i][j] != '*')
                dp[i][j] = (dp[i][j-1]%MOD + dp[i-1][j]%MOD)%MOD;
        }
    }
    cout << dp[n-1][n-1]%MOD;
    return 0;
}

File: summer_school/turtle.py
import math
n = int(input())
MOD = 10e9 + 7
inf = -1000000000000000000000000000000

matrix = [[1 for _ in range(n)] for _ in range(n)]

for j in range(n):
    row = input()
    for u in range(n):
        if row[u] == ".":
            matrix[j][u] = 1
        else:
            matrix[j][u] = None

dp = [[1 for i in range(n+1)] for u in range(n+1)]

for i in range(n+1):
    dp[0][i] += dp[0][i-1]
for i in range(n+1):
    dp[i][0] += dp[i-1][0]

for i in range(1, n+1):
    for u in range(1, n+1):
        if dp[i][u-1] == None:
            dp[i][u] = dp[i-1][u]
        elif dp[i-1][u] == None:
            dp[i][u] = dp[i][u-1]
        else:
            dp[i][u] = dp[i-1][u] + dp[i][u-1]

print(dp)